src/stdio/ScalarIO.sac

```diff
@@ -192,130 +192,153 @@ inline void fprint(TermFile &stream, bool n, int mode)
   }
 }
 
-#define SHOW_TERMFILE_S_( typ)                       \
+#define SHOW_TERMFILE_S_( typ, format)               \
 inline void show( typ n)                             \
 {                                                    \
-  fprint(TermFile::stdout, n);                       \
+  TermFile::printf(QUOTE(%format\n), n);             \
 }
 
-SHOW_TERMFILE_S_(int)
-SHOW_TERMFILE_S_(long)
-SHOW_TERMFILE_S_(char)
-SHOW_TERMFILE_S_(float)
-SHOW_TERMFILE_S_(double)
-SHOW_TERMFILE_S_(bool)
-SHOW_TERMFILE_S_(string)
+SHOW_TERMFILE_S_(int, d)
+SHOW_TERMFILE_S_(long, ld)
+SHOW_TERMFILE_S_(string, s)
 
 #ifdef FULLTYPES
-SHOW_TERMFILE_S_(short)
-SHOW_TERMFILE_S_(byte)
-SHOW_TERMFILE_S_(longlong)
-SHOW_TERMFILE_S_(ushort)
-SHOW_TERMFILE_S_(ubyte)
-SHOW_TERMFILE_S_(uint)
-SHOW_TERMFILE_S_(ulong)
-SHOW_TERMFILE_S_(ulonglong)
+SHOW_TERMFILE_S_(short, hd)
+SHOW_TERMFILE_S_(byte, x)
+SHOW_TERMFILE_S_(longlong, lld)
+SHOW_TERMFILE_S_(ushort, hu)
+SHOW_TERMFILE_S_(ubyte, X)
+SHOW_TERMFILE_S_(uint, u)
+SHOW_TERMFILE_S_(ulong, lu)
+SHOW_TERMFILE_S_(ulonglong, llu)
 #endif /* FULLTYPES */
 
-#define PRINT_TERMFILE_S_(typ, format)               \
-inline void print( typ n)                            \
-{                                                    \
-  TermFile::printf("Dimension:  0\nShape    : < >\n");\
-  TermFile::printf( " " QUOTE(%format\n), n);        \
-}
-
-PRINT_TERMFILE_S_(int, d)
-PRINT_TERMFILE_S_(long, ld)
-PRINT_TERMFILE_S_(string, s)
-
-#ifdef FULLTYPES
-PRINT_TERMFILE_S_(short, hd)
-PRINT_TERMFILE_S_(byte, x)
-PRINT_TERMFILE_S_(longlong, lld)
-PRINT_TERMFILE_S_(ushort, hu)
-PRINT_TERMFILE_S_(ubyte, X)
```

Comment:
Right, but I guess you just copy-pasted `x` and `X` from here.

Reply:
Yup. I figured, perhaps wrongly, that the authors of PRINT_xxx had some particular
reason for treating it that way.